lib/scout_apm/plugs/controller_timer.ex

```diff
@@ -54,20 +54,22 @@ defmodule ScoutApm.Plugs.ControllerTimer do
 
   def maybe_mark_error(conn), do: conn
 
-  # Takes a connection, extracts the phoenix controller & action, then manipulates & cleans it up.
-  # Returns a string like "PageController#index"
+  @doc """
+  Takes a connection, extracts the phoenix controller & action, then manipulates
+  & cleans it up.
+
+  Returns a string like "PageController#index"
+  """
+  @spec action_name(Plug.Conn.t()) :: String.t()
   def action_name(conn) do
-    controller_name = conn.private[:phoenix_controller]
     action_name = conn.private[:phoenix_action]
 
-    # a string like "Elixir.TestappPhoenix.PageController#index"
-    "#{controller_name}##{action_name}"
-    # Split into a list
-    |> String.split(".")
-    # drop "Elixir.TestappPhoenix", leaving just ["PageController#index"]
-    |> Enum.drop(2)
-    # Probably just "joining" a 1 elem array, but recombine this way anyway in case of periods
+    conn.private[:phoenix_controller]
+    |> Module.split()
+    |> trim_app_module_name()
     |> Enum.join(".")
+    # Append action
+    |> String.replace_suffix("", "##{action_name}")
```

Comment:
If I remember right, it's a bit weird in pipelines, but isn't this the `<>` operator?

Reply:
Yep, you can do `|> Kernel.<>("##{action_name}")`

Reply:
Interestingly enough under the hood `replace_suffix` does the same operation, for reference: [`replace_suffix`](https://github.com/elixir-lang/elixir/blob/c0ebd7a7a0d8b6281c9b05086b7f2e9e08216233/lib/elixir/lib/string.ex#L1039) and the implementation of [`append_unless_empty`](https://github.com/elixir-lang/elixir/blob/c0ebd7a7a0d8b6281c9b05086b7f2e9e08216233/lib/elixir/lib/string.ex#L1052)

I tend to use `replace_suffix` over the `Kernel.<>` when piping for readability, but if you would like me to change it to `Kernel.<>` i'm happy to do that.

Reply:
I agree, but I think we've got some places we're using `Kernel.<>` and I'd rather maintain the consistency for now. I may go back and update all of those at once in the future.